packages/manager/apps/pci-rancher/src/components/layout-helpers/CreateRancher/CreateRancher.test.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import userEvent from '@testing-library/user-event';
+import CreateRancher, { CreateRancherProps } from './CreateRancher';
+import { render, waitFor, fireEvent } from '../../../utils/test.provider';
+import { rancherMocked } from '../../../_mock_/rancher';
+import { RancherService } from '../../../api/api.type';
+import dashboardTranslation from '../../../public/translations/pci-rancher/dashboard/Messages_fr_FR.json';
+import listingTranslation from '../../../public/translations/pci-rancher/listing/Messages_fr_FR.json';
+import { rancherPlan, rancherVersion } from '@/_mock_/rancher-resource';
+
+const onCreateRancher = jest.fn();
+const onCancelClick = jest.fn();
+
+const setupSpecTest = async (props?: Partial<CreateRancherProps>) =>
+  waitFor(() =>
+    render(
+      <CreateRancher
+        onCreateRancher={onCreateRancher}
+        plans={rancherPlan}
+        versions={rancherVersion}
+        onCancelClick={onCancelClick}
+        hasRancherCreationError={false}
+        {...props}
+      />,
+    ),
+  );
+
+describe('CreateRancher', () => {
+  it("Given that I don't fill the name field, I shouldn't be able to create my Managed Rancher Service (CTA disabled)", async () => {
```

Comment:
There is a library called jest-cucumber that let's you use .feature file and structure you test case like gherkin

https://www.npmjs.com/package/jest-cucumber